samza-kafka/src/main/scala/org/apache/samza/checkpoint/kafka/KafkaCheckpointManager.scala

```diff
@@ -187,92 +174,83 @@ class KafkaCheckpointManager(
     checkpoints.toMap /* of the immutable kind */
   }
 
+  private def getEarliestOffset(topic: String, partition: Partition): (String, String) = {
+    val systemAdmin = getSystemAdmin()
+    val metaDataMap: java.util.Map[String, SystemStreamMetadata] = systemAdmin.getSystemStreamMetadata(Collections.singleton(topic))
+    val checkpointMetadata: SystemStreamMetadata = metaDataMap.get(topic)
+    if(checkpointMetadata == null)
+      throw new SamzaException("Cannot get metadata for system=%s, topic=%s" format (systemName, topic))
+
+    val partitionMetaData = checkpointMetadata.getSystemStreamPartitionMetadata().get(partition);
+    if(partitionMetaData == null)
+      throw new SamzaException("Cannot get partitionMetaData for system=%s, topic=%s" format (systemName, topic))
+
+    val newestOffset = if (partitionMetaData.getNewestOffset == null)
+        "-1"
+      else
+        partitionMetaData.getNewestOffset()
+
+    (partitionMetaData.getOldestOffset, newestOffset)
+  }
 
   /**
-   * Common code for reading both changelog partition mapping and change log
+   * Code for reading checkpoint log
```

Comment:
Minor: (rewrite docs as follows)
`Reads an entry from the checkpoint log and invokes the provided lambda on that entry`

Reply:
sure.